networkmanager/networkmanager.go

```diff
@@ -658,15 +669,64 @@ func (manager *NetworkManager) prepareRuntimeConfig(instanceID, networkID, hostn
 		CapabilityArgs: make(map[string]interface{}),
 	}
 
-	if hostname != "" {
-		aliases = append([]string{hostname}, aliases...)
+	if len(hosts) != 0 {
+		runtimeConfig.CapabilityArgs["aliases"] = map[string][]string{networkID: hosts}
 	}
 
-	if len(aliases) != 0 {
-		runtimeConfig.CapabilityArgs["aliases"] = map[string][]string{networkID: aliases}
+	return runtimeConfig
+}
+
+func (manager *NetworkManager) isHostnameExists(networkID string, hosts []string) error {
+	instances, ok := manager.instancesData[networkID]
+	if !ok {
+		return nil
+	}
+
+	for _, networkInstanceData := range instances {
+		for _, existHostname := range networkInstanceData.hosts {
+			for _, newHostname := range hosts {
+				if existHostname == newHostname {
+					return aoserrors.Errorf("hostname %s already exists", newHostname)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (manager *NetworkManager) prepareHostnameList(networkID string, params NetworkParams) (hosts []string, err error) {
+	hosts = params.Aliases
+	if params.Hostname != "" {
+		hosts = append(hosts, params.Hostname)
+	}
+
+	if params.ServiceID != "" && params.SubjectID != "" {
+		hosts = append(hosts, fmt.Sprintf(
```

Comment:
This line perfectly matches 120 symbols len. No need to break it.

Reply:
Done